test/options_payload_validation.js

```diff
@@ -0,0 +1,280 @@
+
+var JWT = require('jsonwebtoken');
+var _ = require('lodash');
+
+var secret = 'NeverShareYourSecret';
+
+
+var myvariables = require('./options_payload_validation_data');
+var internals = {};
+
+// bring your own validation function
+internals.validate = function (decoded, request, callback) {
+
+    return callback(null, true);
+};
+
+internals.rolePermissions = function (roleCredentials, entity) {
+
+    var acl = {
+        user: ['may be edited by normal user'], // user can only <edit> this content type
+        admin: ['may not be edited by normal user, only by admin', 'may be edited by normal user']
+        // admin can <edit> both content types
+    };
+
+    var hasPermission = _.includes(acl[roleCredentials], entity);
+    return hasPermission;
+};
+
+internals.payload = function (request, reply) {
+
+    var hasPermission = internals.rolePermissions(request.auth.credentials.role, request.payload.type);
+
+    if (hasPermission) {
+
+        return reply.continue();
+    }
+
+    return reply(Boom.unauthorized('Role may not edit list'));
+
+};
+
+var server = new Hapi.Server();
+server.connection();
+
+server.register(require('../'), function (err) {
+
+    server.auth.strategy('jwt', 'jwt', {
+        key: secret,
+        validateFunc: internals.validate,
+        verifyOptions: { algorithms: ['HS256'] },
+
+        payload: internals.payload
+    });
+
+    server.route([
+        {
+            method: "GET", path: "/",
+            config: {
+                auth: false
+            },
+            handler: function (request, reply) {
+                reply({ text: 'Token not required' });
+            }
+        },
+
+        {
+            method: 'GET', path: '/restricted',
+            config: {
+                auth: 'jwt'
+            },
+            handler: function (request, reply) {
+                reply({ text: 'You used a Token!' })
+                    .header("Authorization", request.headers.authorization);
+            }
+        },
+        {
+            method: 'PUT', path: '/restricted/payloadRequired',
+            config: {
+                auth: {
+                    strategies: ['jwt']
+                },
+            },
+            handler: function (request, reply) {
+                reply({ text: 'You used a Token!', payload: request.payload })
+                    .header("Authorization", request.headers.authorization);
+            }
+        },
+        {
+            method: 'PUT', path: '/restricted/payloadRequired/authDependsOnPayload',
+            config: {
+                auth: {
+                    strategies: ['jwt'],
+                    payload: "required"
+                }
+            },
+            handler: function (request, reply) {
+                reply({ text: 'You used a Token!', payload: request.payload })
+                    .header("Authorization", request.headers.authorization);
+            }
+        }
+    ]);
+
+});
+
+
+test('1', function (t) {
```

Comment:
there are a couple of _duplicate_ (_redundant_) tests in this file, would you mind if I streamlined it?